code/modules/clothing/neck/necklace/pendant_setting.dm

```diff
@@ -0,0 +1,45 @@
+// Pendant with a setting for a stone.
+/obj/item/pendant/setting
+	abstract_type = /obj/item/pendant/setting
+
+/obj/item/pendant/setting/Initialize(ml, material_key)
+	. = ..()
+	// Similar to rings, this allows for mapping pendants with gems.
```

Comment:
yeah i really think it'd be fine to move this up to /obj/item-level decoration handling? idk